pkg/service/backup/service.go

```diff
@@ -817,23 +818,27 @@ func (s *Service) Backup(ctx context.Context, clusterID, taskID, runID uuid.UUID
 			return w.Snapshot(ctx, hi, target.SnapshotParallel)
 		},
 		StageAwaitSchema: func() error {
-			clusterSession, err := s.clusterSession(ctx, clusterID, "")
+			descSchemaHost := liveNodes[0].Addr
+			session, err := s.clusterSession(ctx, clusterID, descSchemaHost)
 			if err != nil {
-				w.Logger.Info(ctx, "No CQL cluster session, backup of schema as CQL files will be skipped", "error", err)
-				return nil
+				if errors.Is(err, cluster.ErrNoCQLCredentials) {
+					w.Logger.Error(ctx, "No CQL cluster credentials, backup of schema as CQL files will be skipped", "error", err)
+					return nil
+				}
+				return errors.Wrapf(err, "create single host (%s) session to", descSchemaHost)
 			}
-			defer clusterSession.Close()
+			defer session.Close()
 
-			if err := w.AwaitSchemaAgreement(ctx, clusterSession); err != nil {
-				return errors.Wrap(err, "await schema agreement")
+			if err := query.RaftReadBarrier(session); err != nil {
```

Comment:
Isn't is just for Scylla's that support RAFT ?
Please leave the old approach for older versions of Scylla.

Reply:
That's true!
This does not break anything for non raft schema clusters in principle, because:
- the raft read barrier is correct CQL stmt in older scylla versions
- the previously used await schema procedure is bugged
- this schema wasn't safe to use with restore anyway

But I can change it to use the driver await schema method (not bugged) and still await the schema agreement just to continue the old behavior.